corehq/apps/cloudcare/static/cloudcare/js/utils.js

```diff
@@ -338,71 +366,64 @@ hqDefine('cloudcare/js/utils', [
         return inputDate;
     };
 
-    var initDateTimePicker = function ($el, extraOptions) {
+    var initDatePicker = function ($el, selectedDate, dateFormat) {
         if (!$el.length) {
             return;
         }
 
-        extraOptions = extraOptions || {};
-        $el.datetimepicker(_.extend({
+        dateFormat = dateFormat || "MM/DD/YYYY";
+        let formats = _.union([dateFormat], ["MM/DD/YYYY", "MM/DD/YY", "YYYY-MM-DD"]);
+        $el.datetimepicker({
+            date: selectedDate,
             useCurrent: false,
             showClear: true,
             showClose: true,
             showTodayButton: true,
             debug: true,
-            extraFormats: ["MM/DD/YYYY", "MM/DD/YY"],
+            format: dateFormat,
+            extraFormats: formats,
+            useStrict: true,
             icons: {
                 today: 'glyphicon glyphicon-calendar',
             },
-            tooltips: {     // use default text, but enable translations
-                today: gettext('Go to today'),
-                clear: gettext('Clear selection'),
-                close: gettext('Close the picker'),
-                selectMonth: gettext('Select Month'),
-                prevMonth: gettext('Previous Month'),
-                nextMonth: gettext('Next Month'),
-                selectYear: gettext('Select Year'),
-                prevYear: gettext('Previous Year'),
-                nextYear: gettext('Next Year'),
-                selectDecade: gettext('Select Decade'),
-                prevDecade: gettext('Previous Decade'),
-                nextDecade: gettext('Next Decade'),
-                prevCentury: gettext('Previous Century'),
-                nextCentury: gettext('Next Century'),
-                pickHour: gettext('Pick Hour'),
-                incrementHour: gettext('Increment Hour'),
-                decrementHour: gettext('Decrement Hour'),
-                pickMinute: gettext('Pick Minute'),
-                incrementMinute: gettext('Increment Minute'),
-                decrementMinute: gettext('Decrement Minute'),
-                pickSecond: gettext('Pick Second'),
-                incrementSecond: gettext('Increment Second'),
-                decrementSecond: gettext('Decrement Second'),
-                togglePeriod: gettext('Toggle Period'),
-                selectTime: gettext('Select Time'),
+            tooltips: dateTimePickerTooltips,
+            parseInputDate: function (dateString) {
+                if (!moment.isMoment(dateString) || dateString instanceof Date) {
+                    dateString = convertTwoDigitYear(dateString);
```

Comment:
I think there's some weird behavior happening here. `convertTwoDigitYear` returns date separated with "-" (MM-DD-YY) which doesn't strictly follow any of the formats so anything processed through `convertTwoDigitYear` is invalid. As of now, the field seems to only accept MM/DD/YYYY. 

 Did you consider using `moment.parseTwoDigitYear` under ["Parsing two digit years"  ](https://momentjs.com/docs/#/parsing/) which overrides the default parsing behavior of two digit years?

Reply:
Oh wow, good catch, thank you.  I'm a little nervous about modifying that moment behavior, because the consequences of doing so would extend beyond this input.  I can't think of anything that'd break, but I'd rather contain the blast radius of this sort of change.  I will switch it to use `/` as the separator though.

Reply:
fix here: https://github.com/dimagi/commcare-hq/pull/32749